e2e/cypress/integration/dashboard-transloadit.spec.ts

```diff
@@ -15,105 +9,125 @@ describe('Dashboard with Transloadit', () => {
 
   it('should upload cat image successfully', () => {
     cy.get('@file-input').selectFile('cypress/fixtures/images/cat.jpg', { force:true })
-    cy.get('.uppy-StatusBar-actionBtn--upload').click()
-
-    cy.wait('@assemblies')
-    cy.wait('@resumable')
 
-    cy.get('.uppy-StatusBar-statusPrimary').should('contain', 'Complete')
+    cy.get('.uppy-StatusBar-actionBtn--upload').click().then(() => {
+      cy.wait(['@assemblies', '@resumable']).then(() => {
+        cy.get('.uppy-StatusBar-statusPrimary').should('contain', 'Complete')
+      })
+    })
   })
 
   it('should close assembly polling when cancelled', () => {
-    cy.get('@file-input').selectFile(['cypress/fixtures/images/cat.jpg', 'cypress/fixtures/images/traffic.jpg'], { force:true })
-    cy.get('.uppy-StatusBar-actionBtn--upload').click()
-
     cy.intercept({
       method: 'GET',
       url: '/assemblies/*',
     }).as('assemblyPolling')
     cy.intercept(
-      { method: 'PATCH', pathname: '/files/*', times: 1 },
-      { statusCode: 204, body: {} },
-    )
-    cy.intercept(
-      { method: 'DELETE', pathname: '/resumable/files/*', times: 1 },
+      { method: 'DELETE', pathname: '/assemblies/*', times: 1 },
       { statusCode: 204, body: {} },
-    )
-    cy.wait('@assemblyPolling')
-    cy.window().then(({ uppy }) => {
-      expect(Object.values(uppy.getPlugin('Transloadit').activeAssemblies).every((a: any) => a.pollInterval)).to.equal(true)
-    })
-    cy.get('button[data-cy=cancel]').click()
+    ).as('delete')
 
     cy.window().then(({ uppy }) => {
-      expect(Object.values(uppy.getPlugin('Transloadit').activeAssemblies).some((a: any) => a.pollInterval)).to.equal(false)
+      cy.get('@file-input').selectFile(
+        ['cypress/fixtures/images/cat.jpg', 'cypress/fixtures/images/traffic.jpg', 'cypress/fixtures/images/car.jpg'],
+        { force:true },
+      ).then(() => {
+        cy.get('.uppy-StatusBar-actionBtn--upload').click().then(() => {
+          cy.wait(['@createAssemblies']).then(() => {
+            expect(Object.values(uppy.getPlugin('Transloadit').activeAssemblies).every((a: any) => a.pollInterval)).to.equal(true)
+
+            uppy.cancelAll()
+
+            cy.wait(['@delete']).then(() => {
+              expect(Object.values(uppy.getPlugin('Transloadit').activeAssemblies).some((a: any) => a.pollInterval)).to.equal(false)
+            })
+          })
+        })
+      })
     })
   })
 
-  it('should emit one assembly-cancelled event when cancelled', FLAKY, () => {
+  // Too flaky at the moment. Arguably, this is not the right place
+  // as this is doing white box testing (testing internal state).
+  // But E2e is more about black box testing, you don’t care about the internals, only the result.
+  // May make more sense to turn this into a unit test.
+  it.skip('should emit one assembly-cancelled event when cancelled', () => {
```

Comment:
Can we skip it only in CI? Since it works fine locally.

Reply:
I don't think cypress offers that option. Only thing I can think of is having some sort of env var, but not sure about that approach

Reply:
> You may also skip at runtime using this.skip(). If a test needs an environment or configuration which cannot be detected beforehand, a runtime skip is appropriate.

I think we can check for `'GITHUB_REF' in process.env`. If that turns out to be more difficult, let’s give up on this idea and let’s stick with `.skip`.

Reply:
I think the better approach is to refactor this test to a unit test.